lib/stats/DefaultStatsFactoryPlugin.js

```diff
@@ -98,27 +287,90 @@ const { makePathsRelative, parseResource } = require("../util/identifier");
+/**
+ * @template {any} T
+ * @template {Object} O
+ * @typedef {Record<string, (object: O, data: T, context: UsualContext, options: UsualOptions, factory: StatsFactory) => void>} ExtractorsByOption
+ */
 
 /**
  * @typedef {Object} SimpleExtractors
- * @property {ExtractorsByOption<Compilation>} compilation
- * @property {ExtractorsByOption<ExtendedAsset>} asset
- * @property {ExtractorsByOption<ExtendedAsset>} asset$visible
- * @property {ExtractorsByOption<{ name: string, chunkGroup: ChunkGroup }>} chunkGroup
- * @property {ExtractorsByOption<Module>} module
- * @property {ExtractorsByOption<Module>} module$visible
- * @property {ExtractorsByOption<Module>} moduleIssuer
- * @property {ExtractorsByOption<ModuleProfile>} profile
- * @property {ExtractorsByOption<ModuleGraphConnection>} moduleReason
- * @property {ExtractorsByOption<Chunk>} chunk
- * @property {ExtractorsByOption<OriginRecord>} chunkOrigin
- * @property {ExtractorsByOption<WebpackError>} error
- * @property {ExtractorsByOption<WebpackError>} warning
- * @property {ExtractorsByOption<{ origin: Module, module: Module }>} moduleTraceItem
- * @property {ExtractorsByOption<Dependency>} moduleTraceDependency
+ * @property {ExtractorsByOption<Compilation, StatsOutput>} compilation
+ * @property {ExtractorsByOption<StatsAsset, StatsAsset>} asset
+ * @property {ExtractorsByOption<StatsAsset, StatsAsset>} asset$visible
+ * @property {ExtractorsByOption<{ name: string, chunkGroup: ChunkGroup }, StatsChunkGroup>} chunkGroup
+ * @property {ExtractorsByOption<Module, StatsModule>} module
+ * @property {ExtractorsByOption<Module, StatsModule>} module$visible
+ * @property {ExtractorsByOption<ModuleProfile, StatsProfile>} profile
+ * @property {ExtractorsByOption<Module, StatsModuleIssuer>} moduleIssuer
+ * @property {ExtractorsByOption<ModuleGraphConnection, StatsModuleReason>} moduleReason
+ * @property {ExtractorsByOption<Chunk, StatsChunk>} chunk
+ * @property {ExtractorsByOption<OriginRecord, StatsChunkOrigin>} chunkOrigin
+ * @property {ExtractorsByOption<WebpackError, StatsError>} error
+ * @property {ExtractorsByOption<WebpackError, StatsError>} warning
+ * @property {ExtractorsByOption<{ origin: Module, module: Module }, StatsModuleTraceItem>} moduleTraceItem
+ * @property {ExtractorsByOption<Dependency, StatsModuleTraceDependency>} moduleTraceDependency
  */
 
+/**
+ * @typedef {Object} StatsError
+ * @property {string} message
+ * @property {string=} chunkName
+ * @property {boolean=} chunkEntry
+ * @property {boolean=} chunkInitial
+ * @property {string=} file
+ * @property {string=} moduleIdentifier
+ * @property {string=} moduleName
+ * @property {string=} loc
+ * @property {string|number=} chunkId
+ * @property {string|number=} moduleId
+ * @property {any=} moduleTrace
+ * @property {any=} details
+ * @property {any=} stack
+ */
+
+/**
+ * @typedef {
```

Comment:
Not sure of the usefulness of this, but thought to include it based on the stats documentation.